sidecar/sokuji_sidecar/planner.py

```
@@ -0,0 +1,550 @@
+# PCIe traffic) — go straight to the cpu tier.
+_LLAMA_MIN_FIT_FRACTION = 0.5
+
+
+def _llamacpp_variant_row(model, machine: Machine, pin: str | None,
+                          reserved_bytes: int = 0, budget_bytes: int | None = None,
+                          downloaded: set | None = None, *, est_bytes):
+    """Pick (quant, tier) for a llamacpp card.
+
+    pin → that quant unconditionally (user's will; --fit copes with memory).
+    budget known → the LARGEST quant that fits FULLY resident
+        (est_bytes × 1.1 ≤ budget − reserved): a fully-resident smaller quant
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)